features/nfc/nfc/NFCEEPROM.h

```diff
@@ -0,0 +1,160 @@
+public:
+    /**
+     * Construct a NFCEEPROM instance.
+     *
+     * @param[in] driver a pointer to a NFCEEPROMDriver instance
+     * @param[in] queue a pointer to the events queue to use
+     * @param[in] ndef_buffer a bytes array used to store NDEF messages
+     */
+    NFCEEPROM(NFCEEPROMDriver *driver, events::EventQueue *queue, const Span<uint8_t> &ndef_buffer);
+
+    /**
+     * The NFCEEPROM delegate. Users of the NFCEEPROM class need to implement this delegate's methods to receive events.
+     */
+    struct Delegate : NFCTarget::Delegate {
+        /**
+         * The NDEF message erasing request completed.
+         *
+         * @param[in] result NFC_OK or an error code on failure
+         */
+        virtual void on_ndef_message_erased(nfc_err_t result) {}
+
+        /**
+         * The NDEF message writing request completed.
+         *
+         * @param[in] result NFC_OK or an error code on failure
+         */
+        virtual void on_ndef_message_written(nfc_err_t result) {}
+
+        /**
+         * The NDEF message reading request completed.
+         *
+         * @param[in] result NFC_OK or an error code on failure
+         */
+        virtual void on_ndef_message_read(nfc_err_t result) {}
+
+    protected:
+        ~Delegate() {}
+    };
+
+    /**
+     * Initialize the NFC EEPROM
+     *
+     * This method must be called before any other method call.
+     *
+     * @return NFC_OK, or an error.
+     */
+    nfc_err_t initialize();
+
+    /**
+     * Set the delegate that will receive events generated by this EEPROM.
+     *
+     * @param[in] delegate the delegate instance to use
+     */
+    void set_delegate(Delegate *delegate);
+
+private:
+    // Implementation of NFCTarget
+    virtual void write_ndef_message();
+    virtual void read_ndef_message();
+    virtual void erase_ndef_message();
+
+    // Implementation of NFCEEPROMDriver::Delegate
+    virtual void on_session_started(bool success);
+    virtual void on_session_ended(bool success);
+    virtual void on_bytes_read(size_t count);
+    virtual void on_bytes_written(size_t count);
+    virtual void on_size_written(bool success);
+    virtual void on_size_read(bool success, size_t size);
+    virtual void on_bytes_erased(size_t count);
+
+    void handle_error(nfc_err_t ret);
+    void continue_write();
+    void continue_read();
+    void continue_erase();
+
+    enum nfc_eeprom_operation_t {
+        nfc_eeprom_idle,
+
+        nfc_eeprom_write_start_session,
+        nfc_eeprom_write_write_bytes,
+        nfc_eeprom_write_write_size,
+        nfc_eeprom_write_end_session,
+
+        nfc_eeprom_read_start_session,
+        nfc_eeprom_read_read_size,
+        nfc_eeprom_read_read_bytes,
+        nfc_eeprom_read_end_session,
+
+        nfc_eeprom_erase_start_session,
+        nfc_eeprom_erase_erase_bytes,
+        nfc_eeprom_erase_write_size,
+        nfc_eeprom_erase_end_session
+    };
+
+    Delegate *_delegate;
```

Comment:
same name as the inherited field from NFCNDEFCapable

Reply:
Well the field is private